### PR TITLE
Fix passing args in storybook for mantine components

### PR DESCRIPTION
**Pull request by ranquild**
Currently stories are implemented in a way that makes it impossible to re-configure arguments for any non-default story. This PR fixes the issue for all mantine components.

Before:

```
export const ReadOnly = args => <Template {...args} readOnly />

<Canvas>
  <Story name="Read only">
    {ReadOnly}
  </Story>
</Canvas>
```

After:

```
export const ReadOnly = Template.bind({});
ReadOnly.args = {
  readOnly: true
};

<Canvas>
  <Story name="Read only" args={ReadOnly.args}>
    {ReadOnly}
  </Story>
</Canvas>
```

How to verify:
- `yarn storybook`
- Compare with storybook in master for mantine components https://master--61d4abcd959e8c003aa51d4c.chromatic.com/
- In this branch you should be able to open any story with pre-configured args and change them later on

### CODE REVIEW

File: frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.stories.mdx [hunk 1]
```diff
@@ -52,96 +52,105 @@ Checkbox buttons allow users to select a single option from a list of mutually e
 
 ## Examples
 
-export const Template = args => (
+export const DefaultTemplate = args => <Checkbox {...args} />;
+
+export const CheckboxGroupTemplate = args => (
```
Comment (npretto) [inline frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.stories.mdx:57]: Here args are not used and we probably don't want to use them (controls are for the checkbox but here we're showcasing how to group them in a checkbox group).
Do we want to remove the controls or are we fine with some stories having controls not working if it makes the code easier to manage and if it's clear that they shouldn't work?
I think if we wanted to disabled them we'd have to set each control to disabled which is kinda tedious so I'd be fine like this but just wanted to make sure

File: frontend/src/metabase/ui/components/typography/Anchor/Anchor.stories.mdx [hunk 2]
```diff
@@ -49,17 +49,11 @@ The Anchor component allows users to display links with themed styles, and repla
 
 ## Examples
 
-export const Default = args => (
+export const DefaultTemplate = args => (
```
Comment (npretto) [inline frontend/src/metabase/ui/components/typography/Anchor/Anchor.stories.mdx:52]: Anchor stories don't use args

File: frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.stories.mdx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.stories.mdx:57]: I've passed the args. I think it makes sense to have them, e.g. to test different `labelPosition` in a CheckboxGroup.

File: frontend/src/metabase/ui/components/typography/Anchor/Anchor.stories.mdx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/ui/components/typography/Anchor/Anchor.stories.mdx:52]: Fixed

File: frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.stories.mdx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.stories.mdx:57]: Also about

> Do we want to remove the controls or are we fine with some stories having controls not working if it makes the code easier to manage and if it's clear that they shouldn't work?

I've been thinking about this and I'm fine with this for now. We could exclude some `argTypes` later on if it causes trouble.

**Review by oisincoveney — APPROVED** [review]
Reviewed locally and code looks good. 

I think we can have the discussion on the "should the controls work for every story" another time if a problem comes up but imo seems good for now

**Pull request merged**